tools/README.md

```diff
@@ -0,0 +1,61 @@
+# NNCF Tools
+
+## Memory Monitor
+
+Memory Monitor is a tool that can be used to measure python program RAM footprint in time. It supports multiple memory types:
+
+- `MemoryType.RSS`: Resident Set Size is the portion of memory occupied by a process that is held in RAM.
+
+- `MemoryType.RSS_TOP`: The same type of memory as above, but the values are parsed from a table output of the *top* command. Usually, reports the same values as RSS, but with worse resolution.
+
+- `MemoryType.SYSTEM`: This metric is defined as the difference between total system virtual memory and system available memory. Be aware, that this way it is affected by other processes that can change RAM availability. It is advised to call `get_data(memory_from_zero=True)` for this type of memory logging, if one is interested in memory footprint for a certain process. This subtracts the starting memory from all values.
+
+RSS and SYSTEM behave differently when mmap is used, e.g. during OV model loading. RSS will report data which was read with mmap enabled as allocated, however this is not necessarily the case. SYSTEM does not report memory loaded with mmap. So it can be used to analyze "pure" memory usage without contribution of mmap pages which most probably will actually be free, but are reported as allocated by RSS.
+
+It is advised to use `MemoryType.SYSTEM` when analyzing memory of python scripts involving OpenVINO model reading. Also, memory monitor itself allocates some memory itself, especially during figure saving. It is advised to use it for measuring large memory processes.
+
+An example usage of memory monitor tool:
+
+```python
+import gc
+import time
+import numpy as np
+
+from functools import partial
+from pathlib import Path
+from tqdm import tqdm
+
+from memory_monitor import MemoryMonitor, MemoryType
+
+save_dir = Path("memory_logs")
```

Comment:
> Is it expected that in case of system memory usage, there can be negative values on the plot?

It is possible, especially with improper usage https://github.com/nikita-savelyevv/nncf/blob/memory-logger-tool/tools/memory_monitor.py#L145

> Have you noticed the same behavior in clean environment?

I haven't seen any negative values as long as there are no other processes running in the background